redash/worker.py

```diff
@@ -1,68 +1,56 @@
 from __future__ import absolute_import
 from datetime import timedelta
-from random import randint
+from functools import partial
 
 from flask import current_app
+import logging
 
 from celery import Celery
-from celery.schedules import crontab
 from celery.signals import worker_process_init
 from celery.utils.log import get_logger
 
-from redash import create_app, extensions, settings
+from rq import get_current_job
+from rq.decorators import job as rq_job
+
+from redash import create_app, extensions, settings, redis_connection
 from redash.metrics import celery as celery_metrics  # noqa
 
 logger = get_logger(__name__)
 
+job = partial(rq_job, connection=redis_connection)
+
+
+class CurrentJobFilter(logging.Filter):
+    def filter(self, record):
+        current_job = get_current_job()
+
+        record.job_id = current_job.id if current_job else ''
+        record.job_description = current_job.description if current_job else ''
+
+        return True
+
+
+def get_job_logger(name):
+    logger = logging.getLogger('rq.job.' + name)
+
+    handler = logging.StreamHandler()
+    handler.formatter = logging.Formatter(settings.RQ_WORKER_JOB_LOG_FORMAT)
+    handler.addFilter(CurrentJobFilter())
+
+    logger.addHandler(handler)
+    logger.propagate = False
+
+    return logger
+
 
 celery = Celery('redash',
                 broker=settings.CELERY_BROKER,
                 broker_use_ssl=settings.CELERY_SSL_CONFIG,
                 redis_backend_use_ssl=settings.CELERY_SSL_CONFIG,
                 include='redash.tasks')
 
-# The internal periodic Celery tasks to automatically schedule.
-celery_schedule = {
-    'refresh_queries': {
-        'task': 'redash.tasks.refresh_queries',
-        'schedule': timedelta(seconds=30)
-    },
-    'empty_schedules': {
-        'task': 'redash.tasks.empty_schedules',
-        'schedule': timedelta(minutes=60)
-    },
-    'refresh_schemas': {
-        'task': 'redash.tasks.refresh_schemas',
-        'schedule': timedelta(minutes=settings.SCHEMAS_REFRESH_SCHEDULE)
-    },
-    'sync_user_details': {
-        'task': 'redash.tasks.sync_user_details',
-        'schedule': timedelta(minutes=1),
-    },
-    'send_aggregated_errors': {
-        'task': 'redash.tasks.send_aggregated_errors',
-        'schedule': timedelta(minutes=settings.SEND_FAILURE_EMAIL_INTERVAL),
-    }
-}
-
-if settings.VERSION_CHECK:
-    celery_schedule['version_check'] = {
-        'task': 'redash.tasks.version_check',
-        # We need to schedule the version check to run at a random hour/minute, to spread the requests from all users
-        # evenly.
-        'schedule': crontab(minute=randint(0, 59), hour=randint(0, 23))
-    }
-
-if settings.QUERY_RESULTS_CLEANUP_ENABLED:
-    celery_schedule['cleanup_query_results'] = {
-        'task': 'redash.tasks.cleanup_query_results',
-        'schedule': timedelta(minutes=5)
-    }
-
-celery_schedule.update(settings.dynamic_settings.custom_tasks())
 
```

Comment:
Since we're removing so much of the Celery stuff, I think it makes sense to also move the call to 
`celery.on_after_configure.connect` which adds the periodic tasks that were installed by Redash extensions to using RQ. WDYT  @arikfr and @rauchy?

Reply:
Sounds right. Can you take it?

Reply:
Sure can tomorrow, if not I can deliver post-merge.

Reply:
Hm, I forgot that there is a thing about scheduled jobs not being deduplicated automatically, see my [ticket from 3,5 years ago](https://github.com/rq/rq-scheduler/issues/121), have we covered that for other places?

I think it may make sense to use `scheduler.delete()` rather than `scheduler.cancel()` in our schedule function, @rauchy. See https://stackoverflow.com/questions/17537771/where-do-i-register-an-rq-scheduler-job-in-a-django-app for the idea. Alternatively use the job id to explicitly checking if the job has already been scheduled as commented on in https://github.com/rq/rq-scheduler/issues/121#issuecomment-489357064.

Reply:
@jezdez you're right, `job.delete()` (I'm assuming you didn't mean `scheduler.delete()`) makes more sense than `scheduler.cancel()`. This is fixed in 9c73f0a. Thanks!